engine/src/main/java/org/hibernate/search/backend/DeleteByQuerySupport.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.lucene.analysis.TokenStream;
+import org.apache.lucene.analysis.tokenattributes.CharTermAttribute;
+import org.apache.lucene.index.Term;
+import org.apache.lucene.search.BooleanQuery;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.search.TermQuery;
+import org.apache.lucene.search.BooleanClause.Occur;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.NumericUtils;
+import org.hibernate.search.backend.SingularTermQuery.Type;
+import org.hibernate.search.util.impl.ScopedAnalyzer;
+
+/**
+ * This class has means to convert all (by default) supported DeletionQueries back to Lucene Queries and to their
+ * String[] representation and back.
+ *
+ * @author Martin Braun
+ */
+public final class DeleteByQuerySupport {
+
+	private DeleteByQuerySupport() {
+		throw new AssertionError( "can't touch this!" );
+	}
+
+	public static final Map<Integer, ToLuceneQuery> TO_LUCENE_QUERY_CONVERTER;
+	static {
+		{
+			Map<Integer, ToLuceneQuery> map = new HashMap<>();
+
+			map.put( SingularTermQuery.QUERY_KEY, new ToLuceneQuery() {
+
+				@Override
+				public Query build(DeletionQuery deletionQuery, ScopedAnalyzer analyzerForEntity) {
+					SingularTermQuery query = (SingularTermQuery) deletionQuery;
+					if ( query.getType() == Type.STRING ) {
+						try {
+							TokenStream tokenStream = analyzerForEntity.tokenStream( query.getFieldName(), (String) query.getValue() );
+							tokenStream.reset();
+							try {
+								BooleanQuery booleanQuery = new BooleanQuery();
+								while ( tokenStream.incrementToken() ) {
+									String value = tokenStream.getAttribute( CharTermAttribute.class ).toString();
+									booleanQuery.add( new TermQuery( new Term( query.getFieldName(), value ) ), Occur.MUST );
+								}
+								return booleanQuery;
+							}
+							finally {
+								tokenStream.close();
+							}
+						}
+						catch (IOException e) {
+							throw new AssertionError( "no IOException can occur while using a TokenStream " + "that is generated via String" );
```

Comment:
If you really need such an exception, we'd better use `org.hibernate.search.exception.AssertionFailure`